### ISSUE TITLE
ng-change on multiselect  executed without user interaction

### ISSUE THREAD

**Issue by mgalic** [long2know/angular-directives-general#5]
As per documentation  for ng-change

> Evaluate the given expression when the user changes the input

one would assume that ng-change directive applied to multiselect directive would behave the same. But ng-change expression is evaluated once initially regardless of user interaction.

Can this be fixed? Thanks in advance for great directive.

**Comment by long2know** [comment]
I'll take a look this afternoon.  I do modify ngModel in cases where an array is not passed in during initialization, so that could be one case that is hard to eliminate.  But, my thinking is that I can do a compare of what is in ngModel's array to determine if it has to be modified during init.

**Comment by FarhanGhumra** [comment]
Yeah, I was going to ask about how to use multiselect in cascading style. If I have agency list and based on selected multiple agency I need to load agency users.

+1

**Comment by long2know** [comment]
Give this a look if you could.  I generally don't use ng-change with the multiselect since it's too granular.  However, I split out the model setting into a localized object... and then transfer that to ngModel only when there is user interaction.

https://embed.plnkr.co/DDdEHF?autoCloseSidebar&show=preview

There is one caveat - if you want 'complex' models returned in ngModel, there's no real way to handle that without changing ngModel.  So, you'd have to pass in the 'complex' models initially as your selected items to keep things succinct.

Another thought that I had was to track when the popup is opened/closed and if there are changes - and then trigger ngChange only once the dropdown is closed (if there are changes).  This eliminates ngChange being called with every single click.  Another option would be to debounce setting of ngModel..

**Comment by long2know** [comment]
Here's another example that only updates ngModel (thus triggering ngChange) when the selection popup is closed.  It helps to eliminate the repetitive triggering of ngChange.

https://embed.plnkr.co/dkf9A9?autoCloseSidebar&show=preview
